gcloud/datastore/connection.py

```diff
@@ -68,13 +70,16 @@ def _request(self, dataset_id, method, data):
             uri=self.build_api_url(dataset_id=dataset_id, method=method),
             method='POST', headers=headers, body=data)
 
-        if headers['status'] != '200':
-            raise Exception('Request failed. Error was: %s' % content)
+        status = headers['status']
+        if status != '200':
+            message = ('Request failed with status code %s. '
+                       'Error was: %s' % (status, content))
+            raise six.moves.http_client.HTTPException(message)
```

Comment:
We already have it:

https://github.com/GoogleCloudPlatform/gcloud-python/blob/56daf8c248be400f69e8c292743a897ea83c34cb/gcloud/storage/exceptions.py#L158